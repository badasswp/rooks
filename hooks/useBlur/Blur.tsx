import { useEffect } from 'react';
import type { JSX } from 'react';
import useBlur from './';

/**
 * Blur component.
 *
 * This component is used to demonstrate
 * the use of the useBlur hook.
 *
 * @returns {JSX.Element} The Blur component.
 */
const Blur = (): JSX.Element => {
  const [isBlurred, isBlurredRef] = useBlur();

  const handleBlur = () => {
    console.log('Blur event has been triggered');
  }

  useEffect(() => {
    if (isBlurred) {
      handleBlur();
    }
  })

  return (
    <>
      <p data-testid="status">{JSON.stringify(isBlurred)}</p>
      <div data-testid="element" ref={isBlurredRef} style={{ height: 100, border: '1px solid #ccc' }} contentEditable/>
    </>
  );
}

export default Blur;
