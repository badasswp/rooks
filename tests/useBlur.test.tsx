import React from 'react';
import '@testing-library/jest-dom';
import { fireEvent, getByTestId, render } from '@testing-library/react';

import Blur from '../hooks/useBlur/Blur';

describe('<UseBlur />', () => {
  it('should render the component with no issues', () => {
    const { container } = render(<Blur/>);

    expect(container).toMatchSnapshot();
  });

  it('should fire the event passed to the useBlur hook', () => {
    const { container } = render(<Blur />);

    const element = getByTestId(container, 'element');

    element.focus();
    expect(getByTestId(container, 'status').textContent).toBe('false');

    element.blur();
  });
});
