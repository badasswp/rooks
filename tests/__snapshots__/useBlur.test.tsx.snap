// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UseBlur /> should render the component with no issues 1`] = `
<div>
  <p
    data-testid="status"
  >
    false
  </p>
  <div
    contenteditable="true"
    data-testid="element"
    style="height: 100px; border: 1px solid #ccc;"
  />
</div>
`;
